README.md

```diff
@@ -177,46 +168,35 @@ header into the bridging header, and you should be good to go:
 To add, and read documents in Swift, the basics are:
 
 ```swift
-var error: NSError?
-
-let fileManager  = NSFileManager.defaultManager()
-let documentsDir = fileManager.URLsForDirectory(.DocumentDirectory, inDomains: .UserDomainMask)[0]
-	as NSURL
-let storeURL     = documentsDir.URLByAppendingPathComponent("cloudant-sync-datastore")
-let path         = storeURL.path
-
-// Create Datastore Manager
-let manager = CDTDatastoreManager(directory: path, error: &error)
-if let err = error {
-    println("Error creating datastore manager: \(err.localizedDescription)")
-}
+do {
+    let fileManager = NSFileManager.defaultManager()
 
-// Note: data store name must not contain capital letters
-let datastore = manager.datastoreNamed("my_datastore", error: &error)
-if let err = error {
-    println("Error creating datastore: \(err.localizedDescription)")
-}
+    let documentsDir = fileManager.URLsForDirectory(.DocumentDirectory, inDomains: .UserDomainMask).last!
 
-// Create a document
-var rev = CDTDocumentRevision(docId:"doc1")
-var body = [
-    "description": "Buy milk",
-    "completed": false,
-    "type": "com.cloudant.sync.example.task"
-] as NSMutableDictionary
-rev.setBody(body)
+    let storeURL = documentsDir.URLByAppendingPathComponent("cloudant-sync-datastore")
+    let path = storeURL.path
 
-// Save the document to the database
-let revision = datastore.createDocumentFromRevision(rev, error: &error)
-if let err = error {
-    println("Error storing document: \(err.localizedDescription)")
-}
+    let manager = try CDTDatastoreManager(directory: path)
+    let datastore = try manager.datastoreNamed("my_datastore")
 
-// Read a document
-let docId = revision.docId
-let retrieved = datastore.getDocumentWithId(docId, error: &error)
-if let err = error {
-    println("Error retrieving document: \(err.localizedDescription)")
+    // Create a document
+    let rev = CDTDocumentRevision(docId: "doc1")
+    rev.body = ["description":"Buy Milk", "completed": false, "type":"com.cloudant.sync.example.task"]
+
+    // Add an attachment - binary data like a JPEG
+    let att1 = CDTUnsavedFileAttachment(path: "/path/to/image/jpg", name: "cute_cat.jpg", type: "image/jpeg")
+    rev.attachments[att1.name] = att1
+
+    // Save the document to the database
+    let revision = try datastore.createDocumentFromRevision(rev)
+
+    // Read a document
+    let docId = revision.docId
+    let retrieved = try datastore.getDocumentWithId(docId)
+
+
+} catch {
+    print("Encountered an error: \(error)")
 }
 ```
```

Comment:
Perhaps wrap the following lines in the example code above so that they can be read with having to scroll to view them:

https://github.com/cloudant/CDTDatastore/pull/208/files#diff-04c6e90faac2675aa89e2176d2eec7d8R174
https://github.com/cloudant/CDTDatastore/pull/208/files#diff-04c6e90faac2675aa89e2176d2eec7d8R184
https://github.com/cloudant/CDTDatastore/pull/208/files#diff-04c6e90faac2675aa89e2176d2eec7d8R187